app/controllers/alchemy/api/elements_controller.rb

```diff
@@ -9,7 +9,8 @@ class Api::ElementsController < Api::BaseController
     def index
       @elements = Element.accessible_by(current_ability, :index)
       if params[:page_id].present?
-        @elements = @elements.where(page_id: params[:page_id])
+        @elements = @elements.joins(:page_version)
+                      .where(alchemy_page_versions: {page_id: params[:page_id]})
```

Comment:
This would yield some version, but not necessary the public version. Maybe something like: 
`@elements.joins(page: :public_version).where(...)...`